File: 03_FSharp/Workers/Incoming/Behaviour.fs
namespace Workers.Incoming

open System

[<Flags>]
type Behaviour =
    | None = 0uy
    | HighPriority = 1uy
    | LongRunning = 2uy
    | ResourceIntensive = 4uy
    | RequiresAffinity = 8uy
    | Retryable = 16uy
    | ExactlyOnce = 32uy

File: 00_CSharp/Workers/Incoming/Behaviour.cs
using System;

namespace Workers.Incoming;

[Flags]
internal enum Behaviour : byte
{
    None = 0,

    HighPriority      = 1 << 0,
    LongRunning       = 1 << 1,
    ResourceIntensive = 1 << 2,
    RequiresAffinity  = 1 << 3,
    Retryable         = 1 << 4,
    ExactlyOnce       = 1 << 5
}